src/components/ReactFlagsSelect/ReactFlagsSelect.tsx

```diff
@@ -189,9 +198,17 @@ const ReactFlagsSelect: React.FC<Props> = ({
               </span>
               {showSelectedLabel && (
                 <span className={styles.label}>
-                  {getLabel(validSelectedValue)}
+                  {typeof displayLabel === "object"
+                    ? displayLabel.primary
+                    : displayLabel}
                 </span>
               )}
+              {showSecondarySelectedLabel &&
+                typeof displayLabel === "object" && (
```

Comment:
The customLabel object should have the secondary value as optional, let's check if the secondary value is available before rendering this span.
Also the object check should use the utility function `isCustomLabelObject(displayLabel)`